ui/rdb/components/SceneGraphInspectorPanel/NodeDetailView.svelte

```diff
@@ -130,6 +126,62 @@
         };
     }
 
+    function handleResetValueButtonDisplay(element, newValue?) {
+        if (newValue === undefined) {
+            newValue = element.value;
+        }
+
+        const id = element.id;
+        let valueChanged = (newValue !== fields[id].value);
+        if (newValue instanceof Array) {
+            valueChanged = false;
+            for (const [i, item] of newValue.entries()) {
+                if (fields[id].value[i] !== item) {
+                    valueChanged = true;
+                    break;
+                }
+            }
+        }
+        for (const child of element.parentElement.children) {
+            if (child.classList.contains('resetValueButton')) {
+                if (valueChanged) {
+                    child.classList.remove('hide');
+                } else {
+                    child.classList.add('hide');
+                }
+                break;
+            }
+        }
+    }
+
+    function onResetValueButtonClicked() {
+        const elements = [];
+        for (const child of this.parentElement.children) {
+            if (child.classList.contains('fieldValue')) {
+                elements.push(child);
+            };
+        }
+        if (elements.length === 1) {
```

Comment:
Please add comments about WHY the length is 1 or 2. 

What happens if length is 0 or > 2? Should those be handled?

Reply:
Added comments and a console log in case we ever hit it